src/testdir/test_indent.vim

```diff
@@ -300,4 +303,50 @@ func Test_indent_overflow_count2()
   close!
 endfunc
 
+" Test that mouse shape is restored to Normal mode after using "gq" when
+" 'indentexpr' executes :normal.
+func Test_indent_norm_with_gq()
+  CheckFeature mouseshape
+  CheckCanRunGui
+
+  let lines =<< trim END
+    func Indent()
```

Comment:
Somehow I cannot reproduce the problem when using an 'indentexpr' in Vim9 script.